deliver/lib/deliver/runner.rb

```diff
@@ -91,10 +93,10 @@ def submit_for_review
 
     private
 
-    def validate_html(screenshots)
+    def validate_html(screenshots, trailers)
       return if options[:force]
-      return if options[:skip_metadata] && options[:skip_screenshots]
-      HtmlGenerator.new.run(options, screenshots)
+      return if options[:skip_metadata] && options[:skip_screenshots] && options[:skip_trailers]
+      HtmlGenerator.new.run(options, screenshots, trailers)
```

Comment:
what happens with the HTMLGenerator is the user skips one of the types (metadata or screenshots or trailers) ? Does it properly marks the data as being skipped in the preview ?